Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -0,0 +1,381 @@
+        }
+
+        // if already in component mode, end current mode and switch active button to the selected component
+        if (inComponentMode)
+        {
+            AzToolsFramework::ComponentModeFramework::ComponentModeSystemRequestBus::Broadcast(
+                &ComponentModeSystemRequests::EndComponentMode);
+
+            ViewportUi::ViewportUiRequestBus::Event(
+                ViewportUi::DefaultViewportId,
+                &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+                m_switcherId,
+                buttonId);
+        }
+
+        // if the newly active button is not the transform button (no component mode associated), emter component mode 
+        if (buttonId != m_transformButtonId)
+        {
+            AzToolsFramework::ComponentModeFramework::ComponentModeSystemRequestBus::Broadcast(
+                &ComponentModeSystemRequests::AddSelectedComponentModesOfType, component->m_component->GetUnderlyingComponentType());
+        }
+    }
+
+    void ComponentModeSwitcher::OnEditorModeDeactivated(
+        [[maybe_unused]] const ViewportEditorModesInterface& editorModeState, [[maybe_unused]] ViewportEditorMode mode)
+    {
+        if (mode == ViewportEditorMode::Component)
+        {
+            // always want to return to the default transform button upon exiting component mode
+            ViewportUi::ViewportUiRequestBus::Event(
+                ViewportUi::DefaultViewportId,
+                &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+                m_switcherId,
+                m_transformButtonId);
+        }
+    }
+
+    void ComponentModeSwitcher::OnEditorModeActivated(
+        [[maybe_unused]] const ViewportEditorModesInterface& editorModeState, [[maybe_unused]] ViewportEditorMode mode)
+    {
+
+        // if the user enters component mode not using the switcher, the switcher should still switch buttons
+        if (mode == ViewportEditorMode::Component)
+        {
+            for (auto componentData : m_addedComponents)
+            {
+                bool isComponentActive;
+                ComponentModeSystemRequestBus::BroadcastResult(
+                    isComponentActive,
+                    &ComponentModeSystemRequestBus::Events::AddedToComponentMode,
+                    componentData.m_pairId,
+                    componentData.m_component->GetUnderlyingComponentType());
+
+                if (isComponentActive)
+                {
+                    ViewportUi::ViewportUiRequestBus::Event(
+                        ViewportUi::DefaultViewportId,
+                        &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+                        m_switcherId,
+                        componentData.m_buttonId);
+                    m_activeSwitcherComponent = &componentData;
+                }
+            }
+        }
+    }
+
+    void ComponentModeSwitcher::RemoveExclusiveComponents(const AZ::Entity& entity)
```

Comment:
suggestion: As mentioned before maybe this should be called `RemoveDuplicateComponents` unless I'm not 100% getting what this is doing.